src/main/java/org/radarcns/redcap/util/RedCapClient.java

```diff
@@ -124,8 +124,13 @@ private String convertRedCapInputSetToString(Set<RedCapInput> data){
                 LOGGER.info("Successful fetch for record {}", recordId);
             }
             String result = response.body().string();
-            String data = new JSONArray(result).get(REDCAP_RESULT_INDEX).toString();
-            return new ObjectMapper().readValue(data, HashMap.class);
+            JSONArray jsonData = new JSONArray(result);
+            if(jsonData.length() > 0) {
+                return new ObjectMapper().readValue(jsonData.get(REDCAP_RESULT_INDEX).toString(), HashMap.class);
```

Comment:
You may want to specify a TypeReference here since Hashmap is generic and  the mapper does not know the types of the generic classes until they are created at runtime. Something like --

```java
return new ObjectMapper().readValue(jsonData.get(REDCAP_RESULT_INDEX).toString(), new TypeReference<HashMap<String, String>>() {} );
```